library/core/src/ascii/ascii_char.rs

```diff
@@ -515,6 +516,37 @@ impl AsciiChar {
         }
     }
 
+    /// Returns a *character* digit corresponding to given numeric value of
+    /// a digit.
+    ///
+    /// When passed the *number* `0`, `1`, …, `8`, `9`, `10`, …, `34`, `35`,
+    /// returns the *character* `'0'`, `'1'`, …, `'8'`, `'9'`, `'a'`, …, `'y'`,
+    /// `'z'` respectively.
+    ///
+    /// If `d >= 36`, returns `None`.  To get a digit only for `d < 10`, use
+    /// [`digit`](Self::digit) instead.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// #![feature(ascii_char, ascii_char_variants)]
+    /// use core::ascii::Char;
+    ///
+    /// assert_eq!(Some(Char::Digit0), Char::from_digit(0));
+    /// assert_eq!(Some(Char::Digit9), Char::from_digit(9));
+    /// assert_eq!(Some(Char::SmallA), Char::from_digit(10));
+    /// assert_eq!(Some(Char::SmallF), Char::from_digit(15));
+    /// assert_eq!(Some(Char::SmallZ), Char::from_digit(35));
+    /// assert_eq!(None, Char::from_digit(36));
+    /// ```
+    #[unstable(feature = "ascii_char", issue = "110998")]
+    #[inline]
+    pub const fn from_digit(d: u32) -> Option<Self> {
+        const DIGITS: [AsciiChar; 36] =
+            *b"0123456789abcdefghijklmnopqrstuvwxyz".as_ascii().unwrap();
+        if d < 36 { Some(DIGITS[d as usize]) } else { None }
```

Comment:
Is this more efficient than simply `DIGITS.get(d as usize)`? Because to me it seems like there is a nother implicit range check in `DIGITS[]` that could be avoided.

Reply:
`get` isn’t const so it cannot be used here. The compiler should be smart enough to remove the second range check.

Reply:
Got it.